java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -82,6 +86,38 @@ public void onClick(View view) {
 				}
 			}
 		});
+
+		LinearLayout profileTile = (LinearLayout) findViewById(R.id.profileTile);
+		profileTile.setOnClickListener(new View.OnClickListener(){
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (mService.isAppConnected()) {
+					intent = new Intent(MainActivity.this, ProfileActivity.class);
+				} else {
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
+					startActivity(intent);
+				}
+			}
+		});
+
+		LinearLayout addMeddicationTile = (LinearLayout) findViewById(R.id.medicationTile);
+		addMeddicationTile.setOnClickListener(new View.OnClickListener(){
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (mService.isAppConnected()) {
+					intent = new Intent(MainActivity.this, AddMedicationActivity.class);
+				} else {
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
```

Comment:
Same comment here as above, we can clean this up and move to the isAppConnect() condition